src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+            globalCtx->msgCtx.unk_E3EE = 0;
+        }
+
+        func_800800F8(globalCtx, 0x100E, ~0x62, &this->actor, 0);
+        globalCtx->msgCtx.msgMode = 0x37;
+
+        player->actor.posRot.pos.x = this->actor.posRot.pos.x; // x = 990.0f
+        player->actor.posRot.pos.y = this->actor.posRot.pos.y; // y = 205.0f
+        player->actor.posRot.pos.z = this->actor.posRot.pos.z; // z = -1220.0f
+
+        player->currentYaw = player->actor.posRot.rot.y = player->actor.shape.rot.y = this->actor.posRot.rot.y;
+
+        this->reward = 0;
+
+        this->actionFunc = EnFr_Activate;
+    } else if (EnFr_IsAboveAndWithin30xzDist(player, this)) {
+        player->unk_6A8 = &this->actor;
+    }
+}
+
+void EnFr_Activate(EnFr* this, GlobalContext* globalCtx) {
+    // Goes to activated idle
+    if (globalCtx->msgCtx.msgMode == 0xC) {
+        globalCtx->msgCtx.msgMode = 0x37;
+        sEnFrPointers.flags = 1;
+        this->actionFunc = EnFr_ActivateCheckFrogSong;
+    } else if (globalCtx->msgCtx.msgMode == 0x37) { // Goes to Frogs 2 Song
+        sEnFrPointers.flags = 1;
+        this->actionFunc = EnFr_ActivateCheckFrogSong;
+    }
+}
+
+void EnFr_ActivateCheckFrogSong(EnFr* this, GlobalContext* globalCtx) {
+    if (sEnFrPointers.flags == 0xB) {
+
+        // Check if all 6 child songs have been played for the frogs
+        if ((gSaveContext.eventChkInf[13] & 2)          // ZL (Purple Frog)
+            && (gSaveContext.eventChkInf[13] & 4)       // Epona (White Frog)
+            && (gSaveContext.eventChkInf[13] & 0x10)    // Saria (Yellow Frog)
+            && (gSaveContext.eventChkInf[13] & 8)       // Suns (Blue Frog)
+            && (gSaveContext.eventChkInf[13] & 0x20)    // SoT (Red Frog)
+            && (gSaveContext.eventChkInf[13] & 0x40)) { // SoS
+            this->actionFunc = EnFr_TalkBeforeFrogSong;
+            this->songIndex = 6;
+            func_8010B680(globalCtx, 0x40AB, &this->actor);
+        } else {
+            this->songIndex = 0;
+            this->actionFunc = EnFr_InitAnimation;
+        }
+    }
+}
+
+void EnFr_InitAnimation(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frog;
+    s32 frogIndex;
+
+    for (frogIndex = 0; frogIndex < 5; frogIndex++) {
+        frog = sEnFrPointers.frogs[frogIndex];
+        if (frog != NULL && (frog->actionFunc == EnFr_ChooseJumpFromLogSpot)) {
+            continue;
+        } else {
+            return;
+        }
+    }
+    func_8010BD58(globalCtx, 0x30);
+    this->actionFunc = EnFr_ListeningToOcarinaNotes;
+}
+
+void EnFr_ListeningToOcarinaNotes(EnFr* this, GlobalContext* globalCtx) {
+    this->songIndex = 7; // No Song
+
+    // Ocarina Song played
+    switch (globalCtx->msgCtx.unk_E3EE - 1) {
+        case 6: // Zeldas Lullaby   (unk_E3EE == 7)
+            this->songIndex = 0;
+            break;
+        case 5: // Epona's Song     (unk_E3EE == 6)
+            this->songIndex = 1;
+            break;
+        case 4: // Saria's Song     (unk_E3EE == 5)
```

Comment:
if you add one to all of the cases you can remove the -1 from the switch check. this is just something the compiler did on its own to make things 0 indexed 
after which you can remove the comments